pandas/_libs/algos.pyx

```diff
@@ -180,6 +180,8 @@ def is_lexsorted(list_of_arrays: list) -> bint:
                 else:
                     result = False
                     break
+            if not result:
+                break
```

Comment:
This makes sense to me. But probably even better instead of the double break, to simply return when we know it's `False`. And we don't even need the `result` variable, since the return at the end of the function should always be `return True`.

```suggestion
                    free(vecs)
                    return False
```

Reply:
Good to know, I was wondering if there was a Cython limitation. Thanks for trying it.